src/image/ExposureInfo.cc

```diff
@@ -160,8 +160,12 @@ ExposureInfo::_startWriteFits(afw::geom::Point2I const & xy0) const {
     data.metadata->combine(getMetadata());
 
     // In the future, we might not have exactly three image HDUs, but we always do right now,
-    // so 1=primary, 2=image, 3=mask, 4=variance, 5+=archive
-    data.metadata->set("AR_HDU", 5, "HDU containing the archive used to store ancillary objects");
+    // so 0=primary, 1=image, 2=mask, 3=variance, 4+=archive
+    //
+    // Historically the AR_HDU keyword was 1-indexed (see RFC-304), and to maintain file compatibility
+    // this is still the case so we're setting AR_HDU to 5 == 4 + 1
+    //
+    data.metadata->set("AR_HDU", 5, "One more than HDU containing the archive used to store ancillary objects");
```

Comment:
I discussed this with @TallJimbo while writing the code.  The code is 0-indexed, the only thing that isn't is the `AR_HDU` keyword in the files.  I could complicate the code to handle backwards compatibility, but it seemed simpler to leave the files alone.  That way old and new versions of afw can both read the old and new files.  I'll give him another chance to weigh in, but I still feel that this is the correct solution.